knowledge-base/kb-richtextbox-auto-text-implementation.md

```diff
@@ -0,0 +1,92 @@
+---
+title: RadRichTextBox AutoText implementation
+description: Shows how you can create reusable text snippets.
+type: how-to
+page_title: RadRichTextBox AutoText implementation
+slug: kb-richtextbox-auto-text-implementation
+position: 0
+tags: autotext, reusable, snippets, richtextbox
+ticketid: 1465065
+res_type: kb
+---
+
+## Environment
+<table>
+    <tbody>
+	    <tr>
+	    	<td>Product Version</td>
+	    	<td>2020.1.508</td>
+	    </tr>
+	    <tr>
+	    	<td>Product</td>
+	    	<td>RadRichTextBox for WPF</td>
+	    </tr>
+    </tbody>
+</table>
+
+
+## Description
+
+In this article, you will find how to implement custom functionality for reusable text snippets (AutoText) in RadRichTextBox.
+
+## Solution
+
+This functionality can be achieved by tracking the keyboard input by attaching to the RadRichTextBox` KeyDown and KeyUp events. When the events are fired you can store and compare the entered text and when it matches a predefined pattern to use the RadDocument`s [CaretPosition](%slug radrichtextbox-features-positioning%) to make a [Selection](%slug radrichtextbox-features-selection%) that will be overwritten with the desired text snippet.
```

Comment:
'by' repeated twice in the first sentence. The second one is pretty long and hard to understand